torch/csrc/autograd/engine.cpp

```diff
@@ -266,6 +283,14 @@ auto Engine::thread_init(int device) -> void {
   // arbitrarily picked to colocate devices.  Maybe the other approach is
   // better.
   set_device(device);
+  // assign each device other than CPU a ready queue and store it in thread local
+  local_ready_queue = std::make_shared<ReadyQueue>();
```

Comment:
Looks like you are assigning ready queues unconditionally here to me! Is the comment wrong or should this be in the if statement?

Reply:
(Meta comment: whenever we assign over the TLS, I want to make sure we're not accidentally "losing" an old `local_ready_queue`. Asserts that `!local_ready_queue` whenever we do this assignment would help build confidence here.)